pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/utils/AggregationUtils.java

```diff
@@ -90,31 +90,31 @@ private static Object mergeMax(@Nullable Object agg, @Nullable Object value) {
   }
 
   @Nullable
-  private static Boolean mergeBoolAnd(@Nullable Object agg, @Nullable Object value) {
+  private static Object mergeBoolAnd(@Nullable Object agg, @Nullable Object value) {
     if (agg == null) {
-      return (Boolean) value;
+      return value;
```

Comment:
is this logic correct? I may not have the context, but it looks like this method tries to implement the SQL and logic. In that case `null and false` is false, but `null and true` is null. We are not enforcing the later here.

Reply:
Good catch. Let me fix it